packages/adblocker/src/data-view.ts

```diff
@@ -87,8 +90,7 @@ export function sizeOfASCII(str: string): number {
  * Return number of bytes needed to serialize `str` UTF8 string.
  */
 export function sizeOfUTF8(str: string): number {
-  const encodedLength = encode(str).length;
-  return encodedLength + sizeOfLength(encodedLength);
+  return 4 + TEXT_ENCODER.encode(str).length;
```

Comment:
`.encode` is considered slower than `.encodeInto`, should we approximate the buffer size then get the length of bytes written? If so, how much it'll take to allocate a few bytes of memory every time. Should we have a shared buffer that can contain pretty much of strings?

Reply:
Unless it is really a hot path, I would keep it as is.

If we need to optimize further, I assume looping over the string and summing up the individual counts (looking at the charCodeAt) could be the fastest:
https://en.wikipedia.org/wiki/UTF-8#Description

But getting this correct for all possible string is still some work. Not sure if it is worth here. But it would avoid all memory allocations or copies.